src/basic/cgroup-util.c

```diff
@@ -2211,6 +2212,62 @@ int cg_trim_everywhere(CGroupMask supported, const char *path, bool delete_root)
         return 0;
 }
 
+int cg_mask_to_string(CGroupMask mask, char **ret) {
+        _cleanup_strv_free_ char **t = NULL;
+        CGroupController c;
+        char *s;
+
+        assert(ret);
+
+        if (!mask)
+                return 0;
+
+        for (c = 0; c < _CGROUP_CONTROLLER_MAX; c++)
+                if (mask & CGROUP_CONTROLLER_TO_MASK(c)) {
+                        char *n;
+                        int r;
+
+                        n = strdup(cgroup_controller_to_string(c));
+                        if (!n)
+                                return -ENOMEM;
+
+                        r = strv_push(&t, n);
+                        if (r < 0)
+                                return r;
+                }
+
+        s = strv_join(t, NULL);
+        if (!s)
+                return -ENOMEM;
```

Comment:
Sure I'll rework it to make shorter and prettier. BTW, I think using an pre-allocated character array on the stack would be even simpler, after all the number of controllers is reasonably low.